### DISCUSSION TITLE
ERROR: HTTP 404 Not Found

### DISCUSSION THREAD

**Discussion by ncars** [porrey/Virtual-ZPL-Printer#42] [Q&A]
Hi.
I've downloaded the Virtual ZPL Printer for testing barcode print. But when testing anything from the ERP system connected to the printer or even just doing a test-page, I get this error message: ERROR: HTTP 404 Not Found.
I've no idea on where to start, any idea? 
Thanks a lot!
![image](https://github.com/porrey/Virtual-ZPL-Printer/assets/150795498/e7242ae6-0eca-4ea6-9962-1c5df188f763)
[Virtual ZPL printer issue.pdf](https://github.com/porrey/Virtual-ZPL-Printer/files/13347602/Virtual.ZPL.printer.issue.pdf)

**Comment by porrey** [comment]
I checked the latest installer and the latest code base to ensure nothing changed or stopped working as expected. I did not have any issues.

The application uses a REST API at **http://api.labelary.com/v1/printers**. Ensure you can access this path.

Documentation on service: https://labelary.com/service.html

**Comment by ncars** [comment]
Hi Daniel.

Thanks so much for your answer.

If I click on that path you have inserted, I get same error message.
What could this mean, do you have any idea?

[cid:0b58ecae-0994-45e9-bc6d-39ce0d7aba57]


Best regards

Niklas Carstensen

Warehouse management Specialist – D365 SCM



NDC-Consulting – Runebergs alle 10, 2860 Søborg

Mobile: +45 2825 5636 | ***@***.******@***.***>

CVR: 42933651



________________________________

**Reply by porrey** [reply]
Is there a firewall within your company?

**Comment by ncars** [comment]
Hi.

Thanks for reply.

I found a solution last night. I was not aware of changing this selecting here (since the printer was just started and running)
[cid:856bd2f9-5206-46a0-98fb-2feb85cdb7d2]

Regards


Best regards

Niklas Carstensen

Warehouse management Specialist – D365 SCM



NDC-Consulting – Runebergs alle 10, 2860 Søborg

Mobile: +45 2825 5636 | ***@***.******@***.***>

CVR: 42933651



________________________________

**Comment by markuspirttilahti** [comment]
I might have the same problem with label size change causing a failured connection to Labelary as Niklas had. I cannot see a picture in Niklas' latest message, so don't know how he solved the issue.

I am using version 3.0. My first time using this tool. I was able to create test prints for the existing 2x2 and 4x6 labels within the virtual printer. Using my browser and going to https://api.labelary.com/v1/printers gives me this text: ERROR: HTTP 404 Not Found. It's the same as what the virtual printer displays to me with title "Labelary Error".

Here's my A4 label setup which works in Labelary but doesn't work in the virtual printer:
![image](https://github.com/porrey/Virtual-ZPL-Printer/assets/153720107/e716efd9-2e7e-4440-a638-6a33296da467)

**Comment by porrey** [comment]
I created the label format the same as your image. I was unable to reproduce your issue. I am able to generate many different test labels including some with embedded images. Can you provide more details. Also including the ZPL that is failing will be helpful.

**Reply by markuspirttilahti** [reply]
The ZPL code we are sending from an ERP system to the printer can be found from this [Labelary permalink](https://labelary.com/viewer.html?density=12&quality=grayscale&width=210&height=297&units=mm&index=0&rotation=0&zpl=CT~~CD%2C~CC%5E~CT~%0A%5EXA%0A%5ECI28%0A%0A%5EFX%20Section%201%20-%20License%20plate%20and%20the%20barcode%0A%5EFO40%2C20%5EA0%2C60%5EFDLICENSE%20PLATE%5EFS%0A%5EFO40%2C130%5EA0%2C300%5EFD%24LicensePlateId%24%5EFS%0A%5EFO450%2C450%5EBY10%5EBCN%2C450%2CN%2CN%2CN%2CA%5EFD%24LicensePlateId%24%5EFS%0A%5EFO0%2C1020%2C2%5EGB2650%2C10%2C10%5EFS%0A%0A%5EFX%20Section%202%20-%20PO%20related%20information%0A%5EFO40%2C1100%5EA0%2C180%5EFD%24Qty%24%20%24UnitID%24%20%20%20%20%20%20%24PrintedDate%3Add%2FMM%2Fyyyy%24%5EFS%0A%5EFO40%2C1350%5EA0%2C180%5EFD%24InventBatchId%24%5EFS%0A%5EFO40%2C1600%5EA0%2C180%5EFD%24AccountName%24%5EFS%0A%5EFO0%2C1800%2C2%5EGB2650%2C10%2C10%5EFS%0A%0A%5EFX%20Section%203%20-%20Item%20number%20and%20a%20barcode%0A%5EFO40%2C1900%5EA0%2C180%5EFD%24ItemId%24%5EFS%0A%5EFO550%2C2100%5EBY9%5EBCN%2C300%2CN%2CN%2CN%2CA%5EFD%24ItemId%24%5EFS%0A%5EFO0%2C2500%2C2%5EGB2650%2C10%2C10%5EFS%0A%0A%5EFX%20Section%204%20-%20Po%20number%2C%20width%2C%20item%20name%0A%5EFO40%2C2550%5EA0%2C300%5EFD%24OrderNum%24%5EFS%0A%5EFO40%2C2800%5EA0N%2C200%5ETBN%2C1500%2C3000%5EFD%24ItemName%24%5EFS%0A%5EFO1400%2C3100%5EA0%2C300%5EFD%24ConfigId%24%5EFS%0A%0A%5EXZ).

Sorry to do it like this but the Github text editor changed the code even when using the Code functionality in the editor. The $ elements in the label are obviously filled with actual data.

**Reply by porrey** [reply]
Is the connection from the ERP system something you built or is it via some built-in capability? I am wondering if there is a possibility it is sending more than the ZPL using the failure.

**Reply by markuspirttilahti** [reply]
The ERP system is Microsoft Dynamics 365 SCM, same that Niklas is using according to his signature. It's printing via Document routing agent, i.e. I register the printer into the DRA.

**Comment by porrey** [comment]
This is what I am getting.

<img width="1163" alt="Screenshot 2023-12-14 at 6 40 42 AM" src="https://github.com/porrey/Virtual-ZPL-Printer/assets/12475115/631d6534-c5f8-43fa-a3af-5975c225a43e">

Label definition:

<img width="887" alt="Screenshot 2023-12-14 at 6 42 14 AM" src="https://github.com/porrey/Virtual-ZPL-Printer/assets/12475115/970dae5d-a91e-4246-b2ae-7c85e7bf52cd">

**Reply by markuspirttilahti** [reply]
Okay, thank you for testing it. For some reason I cannot get this to work. I'll report here if I ever figure out the issue.

**Reply by markuspirttilahti** [reply]
I got a successfull label print from the virtual printer when I sent from the ERP system your 4x6 USPS Shipping label example. I'll continue investigating why the label I want to print in A4 size doesn't come out, but fails.

**Comment by ncars** [comment]
Markus,

Do you also get same error when just testing the Virtual ZPL printer with a test label (not from your ERP)?

[cid:6e2a0f5d-12f2-43f1-b402-f631369f0003]


Best regards

Niklas Carstensen

Warehouse management Specialist – D365 SCM



NDC-Consulting – Runebergs alle 10, 2860 Søborg

Mobile: +45 2825 5636 | ***@***.******@***.***>

CVR: 42933651



________________________________

**Comment by pandente** [comment]
Hi, 

I think we found the issue in Version 3.0.0. 

When changing the Label Unit in the Printer configuration to e.g. and also changing the height and/or the width the inches are changed to a decimal number. You can see that when you change the label unit back to "inches". It seems that the error pops up with decimal inches. After manually changing the height and width (in label unit "inches" to integer values, its working. 


regards

**Comment by porrey** [comment]
Sorry, I was unable to produce this issue.
